### ISSUE TITLE
Incorrect description of project-wide testCoroutineDispatcher settings

### ISSUE THREAD

**Issue by SumeraMartin** [kotest/kotest#2693]
The following document contains info about testCoroutineDispatcher for project config.

https://kotest.io/docs/framework/coroutines/test-coroutine-dispatcher.html

```
class ProjectConfig : AbstractProjectConfig() {
  override val testCoroutineDispatcher = true
}
```

But when I try to use the following code then there is no available override for testCoroutineDispatcher. Not sure if this is a bug (missing property) or incorrect documentation.

I'm using 5.0.0 version of the library.

**Comment by sksamuel** [comment]
Yes that's an oversight. Will be corrected in 5.0.1

**Comment by sksamuel** [comment]
@SumeraMartin 5.0.1 is released

**Comment by svenjacobs** [comment]
@sksamuel I believe the `open` keyword is missing from the [testCoroutineDispatcher](https://github.com/kotest/kotest/blob/aac90077fdd37ba61b0cac42e8b238cb2e5ac97b/kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/config/AbstractProjectConfig.kt#L248) property in version 5.0.3. I'm not able to override this property.

**Comment by sksamuel** [comment]
For the vars, you just set them, rather than override them. @svenjacobs

**Comment by svenjacobs** [comment]
Right, but instead of overriding I have to set the property in the `init` block of my instance of `AbstractProjectConfig`.

```kotlin
class ProjectConfig : AbstractProjectConfig() {

    init {
        testCoroutineDispatcher = true
    }
}
```

This is different from the other properties and also does not match the [documentation](https://kotest.io/docs/framework/coroutines/test-coroutine-dispatcher.html).

**Comment by sksamuel** [comment]
The docs are wrong for sure. I'll fix that now.
But none of the other properties allow you to override. They are all vars.
@svenjacobs

**Comment by sksamuel** [comment]
Actually sorry you're right, I was looking at other config objects.
This will be fixed for 5.0.4 and 5.1.0.

**Comment by iainism** [comment]
> Actually sorry you're right, I was looking at other config objects. This will be fixed for 5.0.4 and 5.1.0.

This appears to still be an issue in 5.5.4 (or a regression?), as attempting to use `override var testCoroutineDispatcher = true` [in ProjectConfig as suggested by the documentation](https://kotest.io/docs/framework/coroutines/test-coroutine-dispatcher.html) gives a lint error `'testCoroutineDispatcher' in 'AbstractProjectConfig' is final and cannot be overridden`?
